src/store/options.js

```diff
@@ -1,37 +1,71 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
 import { store } from 'hybrids';
 
 const UPDATE_ACTION_NAME = 'updateOptions';
 
+export function isUpdateOptionsMessage(message) {
+  return message?.action === UPDATE_ACTION_NAME;
+}
+
+export const DNR_IDS = chrome.runtime
+  .getManifest()
+  .declarative_net_request.rule_resources.map((r) => r.id);
+
 const Options = {
   trackerWheelDisabled: false,
   wtmSerpReport: true,
+  dnr: DNR_IDS.reduce((all, rule) => ({ ...all, [rule]: true }), {}),
```

Comment:
I wonder if we can find a better name then "dnr". As I understand from the code, it represents the IDs of the rulesets that are enabled (starting with all rules being enabled).

Also, is the Options object only used as a fallback? If I a disable a rule in the UI, Options.dnr would still have mappings from all known IDs (in the manifest) to `true`, right? So, it would not be used in that case, but options would be loaded from local storage?

Reply:
What other name would you suggest? I don't mind changing it.

It is exactly opposite to what you wrote, the local storage overwrites the manifest default values. This is a clue of the feature.

Reply:
Maybe I have misunderstand the code. So, technically, the values in the `Options` global define the defaults, right? But on extension startup, the local.storage is checked and if there are values present, they will replace the settings in `Option`? So, what `Options` starts with is only relevant for fresh installations? If that it is the case, what would happen if an extension update brings new dnrRule-ID in the manifest? 

Regarding the name, perhaps `activeDnrRules` or `enabledDnrRules` would be an option (being mappings from ruleIds to true/false depending on whether they are in use)? But it could be that I'm misunderstanding the semantics.

Reply:
If would need to choose, I would go with `dnrRules`. As the value is a boolean, the `active` or `enabled` suits more if it would be an array (as it is in the DNR API). 

About the value, the store model has default values in the definition, and they should be in sync with the value of the manifest (for the performance reason, to not overwrite the defaults at a fresh start). We are integrating over the ids and setting the `true` value. When we will switch to `false`, we will update the default value in the options as well.

Reply:
The store models merge data from source and default values (nested objects too), so if we add a new key in the `dnrRules`, its default value will be used, if the local storage will not have that key yet.